src/ComboBoxStateful/index.jsx

```diff
@@ -0,0 +1,668 @@
+}
+
+export default class ComboBoxStateful extends Component
+{
+    static propTypes =
+    {
+        /**
+         *  Display as read-only for IconButton
+         */
+        buttonIsReadOnly     : PropTypes.bool,
+        /**
+         *  Extra CSS class name
+         */
+        className            : PropTypes.string,
+        /**
+         * Placeholder text to show when no dropdown list options
+         */
+        dropdownPlaceholder  : PropTypes.string,
+        /**
+         * Position of the dropdown relative to the text input
+         */
+        dropdownPosition     : PropTypes.oneOf( [ 'top', 'bottom', 'auto' ] ),
+        /**
+         *  Display as hover when required from another component
+         */
+        forceHover           : PropTypes.bool,
+        /**
+         *  Display as error/invalid
+         */
+        hasError             : PropTypes.bool,
+        /**
+         *  Display Button icon as disabled
+         */
+        iconButtonIsDisabled : PropTypes.bool,
+        /**
+         *  Alignment of the icon
+         */
+        iconPosition         : PropTypes.oneOf( [ 'left', 'right' ] ),
+        /**
+         *  Icon type to display
+         */
+        iconType             : PropTypes.oneOf( [
+            'account',
+            'add-circle',
+            'add',
+            'alert',
+            'approved',
+            'arrow',
+            'arrow-up',
+            'arrow-down',
+            'bell',
+            'board',
+            'calendar',
+            'close-circle',
+            'close-thick',
+            'close',
+            'dash',
+            'dashboard',
+            'deactivated',
+            'declined',
+            'delete',
+            'down',
+            'download',
+            'duplicate',
+            'edit-circle',
+            'edit',
+            'ended',
+            'error',
+            'file',
+            'graph',
+            'hide',
+            'info',
+            'inspect',
+            'left',
+            'lightbulb',
+            'link',
+            'loader',
+            'megaphone',
+            'options',
+            'paused',
+            'pending',
+            'preview',
+            'puzzle-piece',
+            'reset',
+            'right',
+            'search',
+            'show',
+            'star-stroke',
+            'star',
+            'sociomantic',
+            'swap',
+            'table',
+            'up',
+            'upload',
+            'validation',
+            'none',
+        ] ),
+        /**
+         *  HTML id attribute
+         */
+        id                : PropTypes.string,
+        /**
+         *  Placeholder text
+         */
+        inputPlaceholder  : PropTypes.string,
+        /**
+         *  Display as disabled
+         */
+        isDisabled        : PropTypes.bool,
+        /**
+         *  Dropdown list allows multiple selection
+         */
+        isMultiselect     : PropTypes.bool,
```

Comment:
This prop can be removed (multi-select behavior is not implemented so this currently does nothing)

Reply:
That said, maybe we should implement it?